### PR TITLE
Ref: linearise logic in the `peers` module (1/..)

### PR DESCRIPTION
**Pull request by niklaslong**
This begins the process of refactoring the code in the `peers` module, in an attempt reduce the control flow complexity and identify deduplication/modularisation opportunities. I'll be favouring smaller PRs in this process to keep conflicts with other changesets to a minimum. 

Commits for this PR are self-contained refactors.

### CODE REVIEW

**Comment by ljedrz** [comment]
Looks reasonable to me so far :detective:.

**Comment by niklaslong** [comment]
Opening for review as is, as these are incremental changes but do risk conflicting with our other planned changes.

File: network/src/peers.rs [hunk 1]
```diff
@@ -534,81 +549,65 @@ impl<N: Network, E: Environment> Peers<N, E> {
...
-            debug!("Dropping connection request from {} (already connected)", peer_ip);
-        }
-        // Ensure the peer is not restricted.
-        else if self.is_restricted(peer_ip).await {
-            debug!("Dropping connection request from {} (restricted)", peer_ip);
+        if !self.can_connect(peer_ip).await {
+            return;
         }
+
         // Spawn a handler to be run asynchronously.
```
Comment (ljedrz) [inline network/src/peers.rs]: this comment should probably be (re)moved now

**Review by ljedrz — APPROVED** [review]
LGTM :+1:

**Comment by howardwu** [comment]
I would prefer to keep the logic as it was originally written.

**Comment by niklaslong** [comment]
@howardwu I've rebased this in case you'd still like to consider it for merging.

**Comment by niklaslong** [comment]
Closing this for now 👍

**Pull request closed without merging**